src/components/Header/Header.tsx

```diff
@@ -1,18 +1,28 @@
 import { useState, useEffect } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
 import Button from 'common/Button/Button';
 import { Logo } from './components/Logo/Logo';
+import { removeUserAction } from 'store/user/actions';
+import { getUserName } from 'store/selectors';
 import './Header.css';
-import { useNavigate, useLocation } from 'react-router-dom';
 
 export const Header = () => {
   const [isLogedIn, setIsLogedIn] = useState(false);
 
   let token = localStorage.getItem('token');
-  const userName = localStorage.getItem('userName');
+  const userName = useSelector(getUserName);
 
   const navigate = useNavigate();
+  const dispatch = useDispatch();
   const location = useLocation();
 
+  const handleLogout = () => {
+    localStorage.removeItem('token');
+    dispatch(removeUserAction());
+    navigate('/login');
+  };
+
   useEffect(() => {
     setIsLogedIn(!!token);
```

Comment:
nit: You might be able to remove this useEffect and use `!!token` as the initial value of the `isLoggedIn` state:

```
const token = localStorage.getItem('token');
const [isLogedIn, setIsLogedIn] = useState(!!token);
```

Haven't tested it though

Reply:
I removed the useState and useEffect.